### PR TITLE
Remove web3

### LINKED ISSUE #8929: Remove Web3
# Problem

web3 is a huge library with lots of dependencies that we only use for big numbers.

## Implementation

Remove usage of web3 from the money.cljs namespace, replace by bn.js in web3-utils, write tests for BN computations

## Acceptance Criteria

- web3 removed
- tests added

### LINKED ISSUE #8954: Remove unused dependencies
https://docs.google.com/spreadsheets/d/1cQ4cyjJ8xy1Li1Ia3kpiVWkXLeCDmIASRvsiAtlgqW8/edit#gid=0

### PR DESCRIPTION
**Pull request by yenda**
fix #8929 
fix #8954

We can finally get rid of web3

##### Functional

- wallet / transactions

### Steps to test

The PR is just removing unused dependencies and replace the last use of web3 to create BigNumbers by using the underlying BigNumber.js library directly. So I don't expect any change of behavior.

Tested on Android simulator

- wallet should display same numbers as nightly

status: ready

### CODE REVIEW

File: src/status_im/utils/money.cljs [hunk 1]
```diff
@@ -30,26 +30,16 @@
 (defn bignumber [n]
   (when n
     (try
-      (.toBigNumber dependencies/web3-prototype (normalize (str n)))
+      (new dependencies/BigNumber (normalize (str n)))
```
Comment (flexsurfer) [inline src/status_im/utils/money.cljs:33]: i remember we had a problem with BigNumber here? is it solved?

Comment (yenda) [reply src/status_im/utils/money.cljs:33]: we overwrote the web3 BigNumber version, that is just using BigNumber directly

**Review by jakubgs — APPROVED** [review]
Removing things is always great.

**Review by flexsurfer — APPROVED** [review]

**Comment by flexsurfer** [comment]
@yenda lets removed all unused then?


react-native-splash-screen
react-native-svg-transformer
react-native-tcp
react-native-udp
assert
asyncstorage-down - i couldn't find how it's used by keycard @dmitryn ?
events
url

**Comment by flexsurfer** [comment]
btw react-native-safe-area-view isn't used as well, but idk why, probably we should start using it

**Comment by yenda** [comment]
@flexsurfer they are all removed except react-native-splash-screen why do you say it is unused? that is the library that sets the splash screen when the app starts

**Comment by flexsurfer** [comment]
ok thanks!

File: mobile_files/DEPENDENCIES.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+used for keycard
+
+## "react-native-svg": "9.7.1"
+
+used for svg icons, mostly collectibles are using svg
+TODO: updating requires a pod update on iOS
+
+## "react-native-svg-transformer": "^0.12.1"
+
+TODO: not sure about this one, probably necessary, not used directly but used as a babel transform
```
Comment (flexsurfer) [inline mobile_files/DEPENDENCIES.md]: babel transform can be removed as well, isn't used

Comment (yenda) [reply mobile_files/DEPENDENCIES.md]: what about collectibles?

Comment (flexsurfer) [reply mobile_files/DEPENDENCIES.md]: `React Native SVG transformer allows you to import SVG files in your React Native project ` but we don't import any svg files, we use webview to render svg collectables by url

Comment (yenda) [reply mobile_files/DEPENDENCIES.md]: ok removed

**Review by vitvly — APPROVED** [review]
Somewhat related - is there a good way to completely cleanup externs.js somehow? Or only manually?

**Comment by pedropombeiro** [comment]
The cleanup commit is going to screw up the merge in Hermes branch so bad...

**Comment by churik** [comment]
@PombeirP so it shouldn't be tested / merged now?

**Comment by pedropombeiro** [comment]
I think I might have a much easier time merging this on top of Hermes than the other way around. But I can discuss with @yenda.

**Comment by churik** [comment]
Ok. in this case we will wait for Hermes build to be merged / tested firstly.
Is it OK?

**Comment by yenda** [comment]
@churik no please test this one

**Comment by flexsurfer** [comment]
@yenda please resolve conflicts

**Comment by pedropombeiro** [comment]
FYI, the Hermes branch is now rebased on this one.

**Comment by ghost** [comment]
`desktop_files/yarn.lock` and `mobile_files/yarn.lock` changed. Pinging @jakubgs, @PombeirP, and @corpetty

**Comment by pedropombeiro** [comment]
Will merge this as it's been tested as part of #8943

**Comment by ghost** [comment]
`desktop_files/yarn.lock` and `mobile_files/yarn.lock` changed. Pinging @jakubgs, @PombeirP, and @corpetty

**Pull request merged**